src/components/Card.js

```diff
@@ -1,54 +1,82 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleCardClick) {
+  constructor({ name, link, _id , isLiked },
+    cardSelector,
+    handleCardClick,
+    handleTrashButtonClick,
+    handleHeartButton) {
     this._name = name;
     this._link = link;
+    this._id = _id;
+    this.isLiked = isLiked;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleTrashButtonClick = handleTrashButtonClick;
+    this._handleHeartButton = handleHeartButton;
+    //this._setLikeStatus = setLikeStatus;
+    //this._cardElement = this._getElement();
   }
 
   _setEventListeners() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeIcon();
+      this._heartButton.addEventListener("click", () => {
+        this._handleHeartButton(this);
       });
 
-    this._cardElement
+      this._cardElement
       .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteCard();
-      });
+      .addEventListener("click", () =>
+        //console.log("delete card in card.js");
+        //this._handleTrashButton(this._id);
+        this._handleTrashButtonClick(this));
 
-    this._cardElement
+      this._cardElement
       .querySelector(".card__image")
       .addEventListener("click", () => {
-        this._handleCardClick(this._name, this._link);
+        this._handleCardClick({ name: this._name, link: this._link });
       });
   }
 
-  _handleLikeIcon() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
-  }
-
-  _handleDeleteCard() {
+  removeCard(cardElement){
     this._cardElement.remove();
+    console.log("removecard method firing");
     this._cardElement = null;
   }
 
-  getView() {
-    this._cardElement = document
-      .querySelector(this._cardSelector)
-      .content.querySelector(".card__list")
-      .cloneNode(true);
+  setLikeStatus(isLiked) {
+    this.isLiked = isLiked;
+    this._renderLikes();
+ }
 
-    this._cardElement.querySelector(".card__title").innerText = this._name;
-    this._cardElement.querySelector(".card__image").src = this._link;
-    this._cardElement.querySelector(".card__image").alt = this._name;
+  _renderLikes() {
+    if(this.isLiked) {
+      this._heartButton.classList.add("card__like-button_active")
+    } else {
+      this._heartButton.classList.remove("card__like-button_active");
+      console.log("like removed from set like status")
+    }
+   }
 
-    this._setEventListeners();
+ _getElement() {
+  return document
+    .querySelector(this._cardSelector)
+    .content.querySelector(".card")
+    .cloneNode(true);
+}
+  getCardElement() {
+    this._cardElement = this._getElement();
+    const cardTitle = this._cardElement.querySelector(".card__title");
+    const cardImage = this._cardElement.querySelector(".card__image");
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.